modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java

```diff
@@ -205,24 +206,33 @@ private void _addManyToOneRelatedObjectEntries(
 
 					if (objectDefinition.isUnmodifiableSystemObject()) {
 						if (FeatureFlagManagerUtil.isEnabled("LPS-183882")) {
-							SystemObjectDefinitionManager
-								systemObjectDefinitionManager =
-									_systemObjectDefinitionManagerRegistry.
-										getSystemObjectDefinitionManager(
-											objectDefinition.getName());
-
-							relatedObjectEntryAtomicReference.set(
-								(Serializable)ObjectEntryDTOConverterUtil.toDTO(
-									systemObjectDefinitionManager.
-										getBaseModelByExternalReferenceCode(
-											systemObjectDefinitionManager.
-												getExternalReferenceCode(
-													primaryKey),
-											objectDefinition.getCompanyId()),
+							Map<String, Object> dtoModel =
+								ObjectEntryValuesUtil.getObjectEntryValues(
+									objectDefinition.getCompanyId(),
 									dtoConverterContext.
 										getDTOConverterRegistry(),
-									systemObjectDefinitionManager,
-									dtoConverterContext.getUser()));
+									objectDefinition.getName(), primaryKey,
+									_systemObjectDefinitionManagerRegistry,
+									dtoConverterContext.getUser());
+
+							if (dtoModel != null) {
+								ObjectField objectField =
+									_objectFieldLocalService.fetchObjectField(
+										objectDefinition.
+											getTitleObjectFieldId());
+
+								User user = dtoConverterContext.getUser();
+
+								dtoModel.put(
+									objectField.getName(),
+									ObjectEntryValuesUtil.getTitleFieldValue(
+										objectField.getBusinessType(),
+										dtoModel.get(objectField.getName()),
+										user.getLanguageId()));
```

Comment:
@seltong I would change the `getTitleFieldValue` method to receive `User` instead of `userLanguageId`, so you can check inside the `getTitleFieldValue` method if `user != null`